### PR TITLE
Feature/1043 1044 - Package Replication improvements

### PR DESCRIPTION
**Pull request by davidjgonzalez**
Combines updated to JCR Package Replication #1043 and #1043

* Closes package after opening to prevent resource leakage
* Improves "replicated by" to default to the actual user that performs the replication. Legacy OSGi configuration available to allow a named user to be assigned for all package replication (leave this value blank to use the real user)

### CODE REVIEW

**Comment by davidjgonzalez** [comment]
@kwin WDYT?

**Comment by kwin** [comment]
Wow, that was fast. Thanks a lot. 

I think you close the `JcrPackage` too soon though. You should close it only once you no longer use the reference towards `JcrPackage`. Apart from that the method `getJcrPackageLastModified(...)` should rather rely on the method `JcrPackage.getPackage().getCreated()` instead of directly accessing nodes/properties as this returns the date when the package has been last built.

File: bundle/src/main/java/com/adobe/acs/commons/replication/status/impl/JcrPackageReplicationStatusEventHandler.java [hunk 1]
```diff
@@ -231,7 +245,7 @@ public final JobResult process(final Job job) {
 
                     if (resources.size() > 0) {
                         replicationStatusManager.setReplicationStatus(resourceResolver,
-                                this.replicatedBy,
+                                StringUtils.defaultIfEmpty(replicatedBy, "Package Replication"),
```
Comment (kwin) [inline bundle/src/main/java/com/adobe/acs/commons/replication/status/impl/JcrPackageReplicationStatusEventHandler.java]: Why do you need a fallback here as well, the property `PROPERTY_REPLICATED_BY` is never null/empty.

File: bundle/src/main/java/com/adobe/acs/commons/replication/status/impl/JcrPackageReplicationStatusEventHandler.java [hunk 2]
```diff
@@ -309,6 +319,10 @@ private boolean containsJcrPackagePath(final String[] paths) {
                 }
             } catch (RepositoryException e) {
                 log.warn("Error checking if the path [ {} ] is a JCR Package.", path);
+            } finally {
+                if (jcrPackage != null) {
+                    jcrPackage.close();
```
Comment (kwin) [inline bundle/src/main/java/com/adobe/acs/commons/replication/status/impl/JcrPackageReplicationStatusEventHandler.java]: this is too early, you are still referencing the jcr package afterwards

**Comment by davidjgonzalez** [comment]
@kwin - FYI - tried this on 6.2 and replication event is null (fromRequest(..)) .. doing this between flights so haven't had a chance to look yet.

Thanks for other comments; will fix when we get to cruising altitude :)

**Comment by davidjgonzalez** [comment]
@kwin so, getting replication action from the osgi event is only supported for grantite replication events, and not the old cq replication topics; might have to do it a bit more manually to handle both cases

**Comment by kwin** [comment]
This PR looks much better now, but I would still recommend to fix the method to get the last built date (and rely internally on `JcrPackage.getPackage().getCreated()`). Right now the modification date on the package rather reflects the time the package definition has last been modified (i.e. via an updated filter) but not when the package has been last rebuilt.

**Comment by davidjgonzalez** [comment]
@kwin date should be fixed; thanks

**Pull request merged**